src/CoreFoundation/CFMessagePort.cs

```diff
@@ -360,6 +360,8 @@ static IntPtr MessagePortCallback (IntPtr local, int msgid, IntPtr data, IntPtr
 			
 			using (var managedData = Runtime.GetNSObject<NSData> (data)) {
 				var result = callback.Invoke (msgid, managedData);
+				// System will release returned CFData
+				result?.DangerousRetain ();
```

Comment:
While this seems like a good Idea the documentation says the following

> If you want the message data to persist beyond this callback, you must explicitly create a copy of data rather than merely retain it; the contents of data will be deallocated after the callback exits.

So I would suggest doing a copy using`NSData FromBytes (IntPtr bytes, nuint size)` and returning that instead.

Reply:
@dalexsoto I think this is the right fix; that part of the documentation refers to the `data` parameter, not the return value from the callback.

Reply:
Yeah, we're returning this CFData as the _return_ from the callback, which Apple will CFRelease under our feet.

The documentation you are referring to is the NSData that gets passed _to_ the callback.

Thanks for checking though :)